modules/libvirtd/domain.nix

```diff
@@ -530,6 +530,13 @@ let
                 '';
               };
 
+              devicesExtraXml = mkOption {
```

Comment:
The option is missing a default value or a null check, otherwise it doesn't work when not set explicitly. That's the reason why the test fails.